federation/source-config-handlers/src/main/java/org/codice/ddf/admin/sources/opensearch/probe/DiscoverOpenSearchSourceProbeMethod.java

```diff
@@ -13,123 +13,76 @@
+    public static final List<String> OPTIONAL_FIELDS = ImmutableList.of(SOURCE_HOSTNAME, PORT, ENDPOINT_URL, SOURCE_USERNAME, SOURCE_USER_PASSWORD);
+    public static final Map<String, String> SUCCESS_TYPES = getCommonSourceSubtypeDescriptions(CREATED_SOURCE);
+    public static final Map<String, String> FAILURE_TYPES = getCommonSourceSubtypeDescriptions(CERT_ERROR, UNKNOWN_ENDPOINT, CANNOT_CONNECT);
+    public static final Map<String, String> WARNING_TYPES = getCommonSourceSubtypeDescriptions(UNTRUSTED_CA);
     public static final List<String> RETURN_TYPES = ImmutableList.of(DISCOVERED_SOURCES);
 
-    private OpenSearchSourceUtils utils;
-
     public DiscoverOpenSearchSourceProbeMethod() {
         super(OPENSEARCH_DISCOVER_SOURCES_ID,
                 DESCRIPTION,
-                REQUIRED_FIELDS,
+                null,
                 OPTIONAL_FIELDS,
                 SUCCESS_TYPES,
                 FAILURE_TYPES,
-                null,
+                WARNING_TYPES,
                 RETURN_TYPES);
-        utils = new OpenSearchSourceUtils();
     }
 
     @Override
     public ProbeReport probe(OpenSearchSourceConfiguration configuration) {
-        ProbeReport probeReport = new ProbeReport(buildMessage(SUCCESS_TYPES,
-                FAILURE_TYPES,
-                WARNING_TYPES,
-                endpointIsReachable(configuration.endpointUrl())));
-        if (probeReport.containsFailureMessages()) {
-            return probeReport;
-        }
-
-        UrlAvailability availability = utils.confirmEndpointUrl(configuration);
-        if (availability == null) {
-            return probeReport.addMessage(buildMessage(SUCCESS_TYPES,
-                    FAILURE_TYPES,
-                    WARNING_TYPES,
-                    UNKNOWN_ENDPOINT));
-        }
-
-        probeReport.addMessage(buildMessage(SUCCESS_TYPES,
-                FAILURE_TYPES,
-                WARNING_TYPES,
-                availability.getAvailabilityResult()));
-        if (probeReport.containsFailureMessages()) {
-            return probeReport;
-        }
-
-        Map<String, Object> probeResult = new HashMap<>();
-        OpenSearchSourceConfiguration openSearchConfig = new OpenSearchSourceConfiguration(
-                configuration);
-        openSearchConfig.endpointUrl(availability.getUrl())
-                .factoryPid(OPENSEARCH_FACTORY_PID)
-                .configurationHandlerId(OPENSEARCH_SOURCE_CONFIGURATION_HANDLER_ID);
-
-        probeResult.put(DISCOVERED_SOURCES, openSearchConfig);
-        return probeReport.addMessage(buildMessage(SUCCESS_TYPES,
-                FAILURE_TYPES,
-                WARNING_TYPES,
-                CONFIG_CREATED))
-                .probeResults(probeResult);
+        String testUrl = configuration.endpointUrl();
+        return testUrl == null ?
+                discoverOpenSearchUrl(configuration.sourceHostName(), configuration.sourcePort(),
```

Comment:
None of this looks correct. `verifyOpenSearchCapabilities` returns back a report with `CONTENT_TYPE, CONTENT, STATUS_CODE`, not a configuration object. Same goes for `discoverOpenSearchUrl `. This logic should look identical to the logic within the DiscoverCswSourceProbeMethod